### PR TITLE
ATOM-17366 Convert Lyshine to use dynamic pass injection

### PR DESCRIPTION
**Pull request by VickyAtAZ**
Changes include:
- Added LyShineFeatureProcessor which provide the override function to do dynamic pass injection.
- Moved the pass template for lyShineChildPass and RttChildPass from code to data files.
- Removed MainPipeline.pass from AutomatedTesting project
- Updated seed list file to add new LyShinePassReqeuest.azasset
- Change LyShinePass::BuildInternal() to LyshinePass::CreateChildPassesInternal to create passes properly
- Removed the script which was used to modify project specific MainPipeline.pass to add lyshine passes offline

Signed-off-by: Qing Tao <55564570+VickyAtAZ@users.noreply.github.com>

Tested with UI editor, Editor game mode with an ui canvas asset component (MainMenu.uicanvas), ui canvas in game launcher

### CODE REVIEW

File: Gems/LyShine/Assets/Passes/LyShinePassRequest.azasset [hunk 1]
```diff
@@ -12,9 +16,10 @@
             {
                 "LocalSlot": "DepthInputOutput",
                 "AttachmentRef": {
-                    "Pass": "DepthPrePass",
+                    "Pass": "HairParentPass",
```
Comment (rgba16f) [inline Gems/LyShine/Assets/Passes/LyShinePassRequest.azasset:19]: Is it a good idea to request insert based on another optional pass?

File: Gems/LyShine/Assets/Passes/LyShineChildPass.pass [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                        "Pass": "Parent",
+                        "Attachment": "DepthInputOutput"
+                    }
+                }
+            ],
+            "PassData": {
+				"$type": "RasterPassData",
+                "DrawListTag": "uicanvas",
+				"PipelineViewTag": "MainCamera"
+			}
```
Comment (invertednormal) [inline Gems/LyShine/Assets/Passes/LyShineChildPass.pass:41]: Some rogue tabs got in here

File: Gems/LyShine/Assets/Passes/LyShinePassRequest.azasset (hunk 1 above)
Comment (invertednormal) [inline Gems/LyShine/Assets/Passes/LyShinePassRequest.azasset:19]: Seems odd to be referencing hair directly in LyShine - is this correct?

File: Gems/LyShine/Assets/Passes/RttChildPass.pass [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                        "Pass": "Parent",
+                        "Attachment": "DepthInputOutput"
+                    }
+                }
+            ],
+            "PassData": {
+				"$type": "RasterPassData",
+                "DrawListTag": "uicanvas",
+				"PipelineViewTag": "MainCamera"
+			}
```
Comment (invertednormal) [inline Gems/LyShine/Assets/Passes/RttChildPass.pass:55]: tabs

**Review by rgba16f — APPROVED** [review]

**Review by invertednormal — APPROVED** [review]
Looks good - just a few minor issues to fix.

File: Gems/LyShine/Assets/Passes/LyShinePassRequest.azasset (hunk 1 above)
Comment (VickyAtAZ) [reply Gems/LyShine/Assets/Passes/LyShinePassRequest.azasset:19]: It's a issue. I will fix it

**Review by invertednormal — APPROVED** [review]

**Pull request merged**